docs/guides/using-org-slugs.mdx

```diff
@@ -0,0 +1,397 @@
+  > component to allow the user to select a valid organization.
+
+  ```tsx {{ filename: 'src/app/orgs/[slug]/page.tsx' }}
+  import { auth } from '@clerk/nextjs/server'
+  import { OrganizationList } from '@clerk/nextjs'
+
+  export default function Home({ params }: { params: { slug: string } }) {
+    const authObject = auth()
+    const orgSlug = authObject.orgSlug
+
+    if (params.slug != orgSlug) {
+      return (
+        <>
+          <p>Sorry, organization {params.slug} is not valid.</p>
+          <OrganizationList
+            hidePersonal={false}
+            hideSlug={false}
+            afterCreateOrganizationUrl="/orgs/:slug"
+            afterSelectOrganizationUrl="/orgs/:slug"
+            afterSelectPersonalUrl="/me"
+          />
+        </>
+      )
+    }
+
+    return <>TODO</>
+  }
+  ```
+
+  ### Rendering Organization-specific content
+
+  Next, you can render organization-specific content, including the organization's name. To do this, you can
+  [customize the Clerk session token](https://clerk.com/docs/backend-requests/making/custom-session-token) to include
+  the organization name:
+
+  1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=sessions).
+  1. In the navigation sidebar, select **Sessions**.
+  1. In the **Customize session token** section, select the **Edit** button.
+  1. In the modal that opens, you can add any claim to your session token that you need. For this guide, add the following:
+
+  ```json
+  {
+    "org_name": "{{org.name}}"
+  }
+  ```
+
+  1. Select **Save**.
+
+  Now you can render the organization name in your page. This example also displays the user's role within the
+  organization (`authObject.orgRole`), which is included in the [default session claims](https://clerk.com/docs/backend-requests/resources/session-tokens#default-session-claims).
+
+  ```tsx {{ filename: 'src/app/orgs/[slug]/page.tsx' }}
+  import { auth } from '@clerk/nextjs/server'
+  import { OrganizationList } from '@clerk/nextjs'
+
+  export default function Home({ params }: { params: { slug: string } }) {
+    const authObject = auth()
+    const orgSlug = authObject.orgSlug
+
+    if (params.slug != orgSlug) {
+      return (
+        <>
+          <p className="pb-8">Sorry, organization {params.slug} is not valid.</p>
+          <OrganizationList
+            hidePersonal={false}
+            hideSlug={false}
+            afterCreateOrganizationUrl="/orgs/:slug"
+            afterSelectOrganizationUrl="/orgs/:slug"
+            afterSelectPersonalUrl="/me"
+          />
+        </>
+      )
+    }
+    let orgName = authObject.sessionClaims['org_name'] as string
+
+    return (
+      <div>
+        {orgName && <h2>Welcome to organization {orgName}</h2>}
+        <p>
+          Your role in this organization: <span>{authObject.orgRole}</span>
+        </p>
+      </div>
+    )
+  }
+  ```
+
+  ### Client-side Routing
```

Comment:
What do you think of moving this example to the same section as `[Server Rendered Organization Page](https://clerk.com/docs/pr/1664/guides/using-org-slugs#server-rendered-organization-page)`

We could update the title, and include two tabs of code snippets between app router and pages router.

I think it's better for developers to make this distinction on the actual code snippet, rather then having to scroll down to a specific section.

Reply:
I like that idea! I'll try it out.

Reply:
90a4442713667c1881425bd6b6c6ba5d5804de28

I like that better!